src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -28,89 +39,1031 @@ const ActorInit En_Js_InitVars = {
+            func_80968B8C(this, play);
+            this->actionFunc = func_80969B5C;
+            func_80968A5C(this);
+            if (func_809692A8(ENJS_GET_TYPE(&this->actor) + 4)) {
+                Actor_MarkForDeath(&this->actor);
+                break;
+            }
+            break;
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+            this->maskDlSelector = ENJS_GET_TYPE(&this->actor) - 4;
+            this->actionFunc = func_8096A104;
+            break;
+    }
+}
+
+void EnJs_Destroy(Actor* thisx, PlayState* play) {
+    u32 paramsF;
+    EnJs* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+
+    paramsF = ENJS_GET_TYPE(&this->actor);
+    switch (paramsF) {
+        case 0:
+            func_80165690();
+            return;
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+            if (!func_809692A8(paramsF)) {
+                func_80969400(ENJS_GET_TYPE(&this->actor));
+            }
+        default:
+            return;
+    }
+}
+
+void func_80968A5C(EnJs* this) {
+    if (Rand_ZeroOne() < 0.5f) {
+        Animation_MorphToLoop(&this->skelAnime, &gMoonChildRunningAnim, -10.0f);
+        this->unk_2B4 = 5.0f;
+    } else {
+        Animation_MorphToLoop(&this->skelAnime, &gMoonChildWalkingAnim, -10.0f);
+        this->unk_2B4 = 2.5f;
+    }
+    this->unk_2B8 &= ~4;
+    this->unk_2BC = Rand_ZeroFloat(40.0f) + 80.0f;
+}
+
+void func_80968B18(EnJs* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gMoonChildStandingAnim, -10.0f);
+    this->unk_2B4 = 0.0f;
+    this->unk_2BC = Rand_ZeroFloat(20.0f) + 40.0f;
+    this->unk_2B8 |= 4;
+}
+
+s32 func_80968B8C(EnJs* this, PlayState* play) {
+    Path* path;
+    f32 f0;
+    f32 f2;
+    s32 params;
+    s32 i;
+    f32 sp18;
+    Vec3s* phi_a0;
+
+    sp18 = 0.0f;
+
+    params = ENJS_GET_PATH_INDEX(&this->actor) >> 0xA;
+    if (params != 0x3F) {
+        this->path = play->setupPathList + params;
+        if (this->path != NULL) {
+            path = this->path;
+
+            phi_a0 = Lib_SegmentedToVirtual(path->points);
+
+            for (i = 0; i < path->count; i++, phi_a0++) {
+                f0 = phi_a0->x - this->actor.world.pos.x;
+                f2 = phi_a0->z - this->actor.world.pos.z;
+                f0 = SQ(f0) + SQ(f2);
+                if (i == 0) {
+                    sp18 = f0;
+                    this->unk_2B0 = 0;
+                } else if (f0 < sp18) {
+                    sp18 = f0;
+                    this->unk_2B0 = i;
+                }
+            }
+
+            this->unk_2B0++;
+            if (this->unk_2B0 >= path->count) {
+                this->unk_2B0 = 0;
+            }
+        }
+    } else {
+        this->path = NULL;
+        this->unk_2B0 = 0;
+    }
+
+    return false;
+}
+
+s32 func_80968CB8(EnJs* this) {
+    Path* path;
+    s32 pad;
+    f32 temp_fa0;
+    f32 temp_fa1;
+    Vec3s* temp_v1;
+
+    if (this->path == NULL)
+        return true;
```

Comment:
```suggestion
    if (this->path == NULL){
        return true;
    }
```
slight nit, I think this looks clearer, but I'll leave it up to you

Reply:
originally i wrote it on a single line, but clang-format messed with it, i'll do it like you suggest